internal/repository/repository.go

```diff
@@ -40,6 +40,10 @@ type EngineRepository interface {
 	RateLimit() RateLimitEngineRepository
 }
 
+type EntitlementsRepository interface {
```

Comment:
Why do we have the `EntitlementsRepository`? Why not have the `TenantLimitRepository` as a direct call in the `ApiRepository` and `EngineRepository` which also gets passed as an argument to the metering struct?

Reply:
I figured it make more sense to separate this responsibility and share it across the two repositories. in the future we'll likely want to add additional things on the entitlement repositories (i.e. feature flags) so i think having it as its own thing make sense